src/storage/mod.rs

```diff
@@ -25,17 +28,17 @@ mod types;
 pub use self::engine::{Engine, Snapshot, Dsn, TEMP_DIR, new_engine, Modify, Cursor,
                        Error as EngineError};
 pub use self::engine::raftkv::RaftKv;
-pub use self::txn::SnapshotStore;
+pub use self::txn::{SnapshotStore, Scheduler, Msg, SchedCh};
 pub use self::types::{Key, Value, KvPair};
-pub type Callback<T> = Box<FnBox(Result<T>) + Send>;
+pub type Callback<T> = Option<Box<FnBox(Result<T>) + Send>>;
 
 pub type CfName = &'static str;
 pub const DEFAULT_CFS: &'static [CfName] = &["default", "lock"];
 
 #[cfg(test)]
 pub use self::types::make_key;
 
-#[derive(Debug)]
+#[derive(Debug, Clone)]
```

Comment:
Will the `Clone` cause more memory allocation when pass `Mutation` around? @BusyJay